pkg/scheduler/framework/v1alpha1/framework.go

```diff
@@ -422,54 +423,59 @@ func (f *framework) RunUnreservePlugins(
 func (f *framework) RunPermitPlugins(
 	pc *PluginContext, pod *v1.Pod, nodeName string) *Status {
 	timeout := maxTimeout
-	statusCode := Success
+	var needsWait bool
 	for _, pl := range f.permitPlugins {
 		status, d := pl.Permit(pc, pod, nodeName)
-		if !status.IsSuccess() {
-			if status.Code() == Unschedulable {
-				msg := fmt.Sprintf("rejected by %v at permit: %v", pl.Name(), status.Message())
-				klog.V(4).Infof(msg)
-				return NewStatus(status.Code(), msg)
-			}
-			if status.Code() == Wait {
-				// Use the minimum timeout duration.
-				if timeout > d {
-					timeout = d
-				}
-				statusCode = Wait
-			} else {
-				msg := fmt.Sprintf("error while running %v permit plugin for pod %v: %v", pl.Name(), pod.Name, status.Message())
-				klog.Error(msg)
-				return NewStatus(Error, msg)
+		if status.IsSuccess() {
+			continue
+		}
+
+		switch status.Code() {
+		case Unschedulable:
+			msg := fmt.Sprintf("rejected by %v at permit: %v", pl.Name(), status.Message())
+			klog.V(4).Infof(msg)
+			return NewStatus(status.Code(), msg)
+		case Wait:
+			if timeout > d {
+				timeout = d
 			}
+			needsWait = true
+		default:
+			msg := fmt.Sprintf("error while running %v permit plugin for pod %v: %v", pl.Name(), pod.Name, status.Message())
+			klog.Error(msg)
+			return NewStatus(Error, msg)
 		}
 	}
 
+	if !needsWait {
```

Comment:
This seems unnecessary. When the code hits this, `needsWait` should always be true. Alternatively, I think you can move the wait pod logic in the `switch` under `case Wait` but it's your call.

Reply:
I'd like to keep the current logic